ethcore/src/executive.rs

```diff
@@ -276,7 +276,7 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 
 			let cost = builtin.cost(data);
 			if cost <= params.gas {
-				builtin.execute(data, &mut output);
+				builtin.execute(data, &mut output)?;
```

Comment:
This can fail without `discarding` or `reverting_to_checkpoint`.

Currently we don't have any state-altering builtins, but might be good to have that in mind. Also I'm not entirely sure if one leaking state snapshot would ever be a problem, but IMHO would be good to address that.